docs/lsm/embedded_entities/embedded_entities.rst

```diff
@@ -147,12 +147,33 @@ Tracking embedded entities across updates
 
 
 Depending on what the embedded entities are modeling, you might want to keep track of which embedded entities
-were added or removed during an update. This section describes how to track embedded entities during the update flow of
-a service. When using the simple lifecycle, this is supported out of the box by wrapping the call to ``lsm::all()`` with the :py:meth:`inmanta_plugins.lsm.insert_removed_embedded_entities<inmanta_plugins.lsm.insert_removed_embedded_entities>` plugin.
+were added or removed during an update, in order to apply custom logic to them. This section describes how to track
+embedded entities during the update flow of a service.
+
+When using the "simple" lifecycle, this is supported out of the box by wrapping the call to ``lsm::all()`` with the
+``insert_removed_embedded_entities`` plugin.
+
+During each step of the update, the ``insert_removed_embedded_entities``
+plugin will compare two sets of attributes (the "current" set and the "previous" set) to determine which embedded entities were added or removed.
+The plugin will accordingly set the following boolean attributes on the relevant embedded entities: ``_removed`` and ``_added``. These values can
+then be used in the model to implement custom logic.
+
+.. note::
+    The "simple" lifecycle defines out-of-the-box which pair of sets should be compared at each step of the update.
+    Please refer to the :ref:`Tracking embedded entities when using a custom lifecycle<using_custom_lifecycle>` section
+    below for more information on how to define which pairs should be compared when using a custom lifecycle.
+
+.. note::
+    To set a different naming scheme for these tracking attributes, use the ``removed_attribute`` and
+    ``added_attribute`` parameters of the ``insert_removed_embedded_entities`` plugin.
+
+
```

Comment:
I think this plugin only works when `strict_modifier_enforcement=true` but I'm not sure. I think it is because embedded entities need an identity to be able to compare them and  `strict_modifier_enforcement` adds that. 

Something to test and document 

On closer inspection, it needs to have '__lsm_key_attributes' set IF it is of arity higher then 1?

(see slack)

Reply:
Added a note regarding `strict_modifier_enforcement`